pulp_rpm/app/viewsets.py

```diff
@@ -250,30 +250,50 @@ def create(self, request):
         serializer = CopySerializer(data=request.data, context={'request': request})
         serializer.is_valid(raise_exception=True)
 
-        source_repo = serializer.validated_data['source_repo']
-        # source_repo_version = serializer.validated_data['source_repo_version']
-        dest_repo = serializer.validated_data['dest_repo']
-        criteria = serializer.validated_data.get('criteria', None)
         dependency_solving = serializer.validated_data['dependency_solving']
-        content_urls = serializer.validated_data.get('content', None)
+        config = serializer.validated_data['config']
 
-        content = []
-        if content_urls:
-            for url in content_urls:
-                c = NamedModelViewSet().get_resource(url, Content)
-                content.append(c.pk)
-
-        source_repos = [source_repo]
-        dest_repos = [dest_repo]
+        config, repos = self._process_config(config)
 
         async_result = enqueue_with_reservation(
-            tasks.copy_content, [*source_repos, *dest_repos],
-            args=[source_repo.latest_version().pk, dest_repo.pk, criteria, content,
-                  dependency_solving],
+            tasks.copy_content, repos,
+            args=[config, dependency_solving],
             kwargs={}
         )
         return OperationPostponedResponse(async_result, request)
 
+    def _process_config(self, config):
+        """
+        Change the hrefs into pks within config.
+
+        This method also implicitly validates that the hrefs map to objects and it returns a list of
+        repos so that the task can lock on them.
+        """
+        result = []
+        repos = []
+
+        for entry in config:
+            r = dict()
+            source_version = NamedModelViewSet().get_resource(entry["source_repo_version"],
+                                                              RepositoryVersion)
+            dest_repo = NamedModelViewSet().get_resource(entry["dest_repo"], RpmRepository)
+            r["source_repo_version"] = source_version.pk
+            r["dest_repo"] = dest_repo.pk
+            repos.extend((source_version.repository, dest_repo))
+
+            if entry.get("dest_repo_version"):
+                r["dest_repo_version"] = (
```

Comment:
We shouldn't name it ```dest_repo_version``` because it shouldn't be an HREF (the resource doesn't exist yet).  It needs to be an integer.  Let's go with ```dest_base_version``` or ```base_version``` for symmetry with the ```modify/``` endpoint.

Reply:
Likewise this resource lookup below doesn't really make sense.